tests/mbt/driver/mbt_test.go

```diff
@@ -0,0 +1,577 @@
+	t.Logf("Number of sent packets: %v", stats.numSentPackets)
+	t.Logf("Number of blocks: %v", stats.numBlocks)
+	t.Logf("Number of transactions: %v", stats.numTxs)
+	t.Logf("Average summed block time delta passed per trace: %v", stats.totalBlockTimePassedPerTrace/time.Duration(numTraces))
+}
+
+func RunItfTrace(t *testing.T, path string) {
+	t.Helper()
+	t.Logf("🟡 Testing trace %s", path)
+
+	// Load trace
+	trace := &itf.Trace{}
+	if err := trace.LoadFromFile(path); err != nil {
+		t.Fatalf("Error loading trace file: %s", err)
+	}
+
+	expectedVarNames := []string{"currentState", "params", "trace"}
+
+	varNames := make(map[string]bool, len(expectedVarNames))
+	// populate the set
+	for _, varName := range trace.Vars {
+		varNames[varName] = true
+	}
+
+	// sanity check: each expected var name should be in the set
+	for _, expectedVarName := range expectedVarNames {
+		_, ok := varNames[expectedVarName]
+		require.True(t, ok, "Expected var name %v not found in actual var names %v", expectedVarName, varNames)
+	}
+	// extra var names are ok, so no need to change the length
+
+	t.Log("Reading params...")
+	params := trace.States[0].VarValues["params"].Value.(itf.MapExprType)
+
+	consumersExpr := params["ConsumerChains"].Value.(itf.ListExprType)
+	initialValSetExpr := params["InitialValidatorSet"].Value.(itf.MapExprType)
+
+	initialValSet := make(map[string]int64)
+	for val, power := range initialValSetExpr {
+		initialValSet[val] = power.Value.(int64)
+	}
+
+	consumers := make([]string, len(consumersExpr))
+	for i, chain := range consumersExpr {
+		consumers[i] = chain.Value.(string)
+	}
+
+	chains := append(consumers, "provider")
+
+	t.Log("Chains are: ", chains)
+
+	// create params struct
+	vscTimeout := time.Duration(params["VscTimeout"].Value.(int64)) * time.Second
+
+	unbondingPeriodPerChain := make(map[ChainId]time.Duration, len(consumers))
+	trustingPeriodPerChain := make(map[ChainId]time.Duration, len(consumers))
+	ccvTimeoutPerChain := make(map[ChainId]time.Duration, len(consumers))
+	for _, consumer := range chains {
+		unbondingPeriodPerChain[ChainId(consumer)] = time.Duration(params["UnbondingPeriodPerChain"].Value.(itf.MapExprType)[consumer].Value.(int64)) * time.Second
+		trustingPeriodPerChain[ChainId(consumer)] = time.Duration(params["TrustingPeriodPerChain"].Value.(itf.MapExprType)[consumer].Value.(int64)) * time.Second
+		ccvTimeoutPerChain[ChainId(consumer)] = time.Duration(params["CcvTimeout"].Value.(itf.MapExprType)[consumer].Value.(int64)) * time.Second
+	}
+
+	modelParams := ModelParams{
+		VscTimeout:              vscTimeout,
+		CcvTimeout:              ccvTimeoutPerChain,
+		UnbondingPeriodPerChain: unbondingPeriodPerChain,
+		TrustingPeriodPerChain:  trustingPeriodPerChain,
+	}
+
+	valExprs := params["Nodes"].Value.(itf.ListExprType)
+	valNames := make([]string, len(valExprs))
+	for i, val := range valExprs {
+		valNames[i] = val.Value.(string)
+	}
+
+	// initialValSet has the right vals, but not yet the right powers
```

Comment:
Why doesn't it have the right voting powers?
Don't we get the right voting powers earlier when we do:
```
		initialValSet[val] = power.Value.(int64)
```